cli/src/semgrep/output.py

```diff
@@ -164,11 +221,13 @@ def __init__(
         self.engine_type: EngineType = EngineType.OSS
 
         self.final_error: Optional[Exception] = None
-        formatter_type = FORMATTERS.get(self.settings.output_format)
-        if formatter_type is None:
-            raise RuntimeError(f"Invalid output format: {self.settings.output_format}")
 
-        self.formatter = formatter_type()
+        self._formatters: Dict[Optional[str], BaseFormatter] = {}
+        for output_destination, output_format in self.settings.get_outputs():
+            formatter_type = FORMATTERS.get(output_format)
+            if formatter_type is None:
+                raise RuntimeError(f"Invalid output format: {output_format}")
+            self._formatters[output_destination] = formatter_type()
```

Comment:
Heads up, my [recent PR](https://github.com/semgrep/semgrep/pull/10113) touches this code a bit, so you'll likely need to rebase. I tried to be mindful about the changes to support multiple output formats, so I think you shouldn't run into issues when rebasing, but let me know if you have questions.